sdk/helper/certutil/helpers.go

```diff
@@ -533,63 +541,118 @@ func StringToOid(in string) (asn1.ObjectIdentifier, error) {
 	return asn1.ObjectIdentifier(ret), nil
 }
 
-// Validates that the combination of keyType, keyBits, and hashBits are
-// valid together; replaces individual calls to ValidateSignatureLength and
-// ValidateKeyTypeLength.
-func ValidateKeyTypeSignatureLength(keyType string, keyBits int, hashBits *int) error {
-	if err := ValidateKeyTypeLength(keyType, keyBits); err != nil {
-		return err
-	}
+// Returns default key bits for the specified key type, or the present value
+// if keyBits is non-zero.
+func DefaultOrValueKeyBits(keyType string, keyBits int) (int, error) {
```

Comment:
We never return an error in any circumstance from this function, could this be simplified?

Reply:
We don't here, but `DefaultOrValueHashBits` returns an err... does it make sense (since this is in the SDK) to keep the signature the same between the two and thus leave the potential for an error open in the future? That was my reasoning for including an (unused) error type.

If we chose to add ed25519 to `defaultAlgorithmKeyBits` we could turn `!present` into an error case, much like we've done with `expectedNISTPCurveHashBits` below. But that'd also be validating algorithm type, which I think should be done later.

Thoughts?